pkg/systemd/services/timer-dialog.jsx

```diff
@@ -89,6 +90,12 @@ const CreateTimerDialogBody = ({ owner }) => {
         <TimePicker className="create-timer-time-picker"
                     time={repeatPatterns[idx].time || "00:00"}
                     is24Hour
+                    isOpen={repeatPatterns[idx].isOpen || false}
+                    setIsOpen={isOpen => {
+                        const arr = [...repeatPatterns];
+                        arr[idx].isOpen = isOpen;
```

Comment:
This does not do what you intend. `...` only creates a copy of the list, but not a deep copy of the elements inside it:
```node
> e1 = {open: false}; e2 = {open: true}; l = [e1, e2]
[ { open: false }, { open: true } ]
> l[0]
{ open: false }
> l2 = [...l]
[ { open: false }, { open: true } ]
> l2[0].open = true
true
> l
[ { open: true }, { open: true } ]
```
So you either need to deep-copy (better), or spare the ceremony and directly change `repeatPatterns[idx].isOpen` (it's bad behaviour, but I don't know how it matters in practice).

Reply:
It is the same pattern as in `onChange` 2 lines down.  Let me take a look at the deep copy for both cases.